### PR TITLE
Pull update.is_valid to python layer

### LINKED ISSUE #2868: Move assert_update_viable into python

### PR DESCRIPTION
**Pull request by sregales-TNO**
**Issue**
Resolves #2868

### CODE REVIEW

File: res/enkf/es_update.py [hunk 1]
```diff
@@ -26,27 +26,26 @@ def setGlobalStdScaling(self, weight):
 
     def smootherUpdate(self, run_context):
```
Comment (sregales-TNO) [inline res/enkf/es_update.py:27]: I am aware that I reduced readability

**Comment by sregales-TNO** [comment]
test ert please!

**Comment by joakim-hove** [comment]
I think the signature of the `is_valid()` function should get some love and attention before it is exported to Python. It is in particular the `analysis_config` argument which I think should be replaced removed. Currently the `analysis_config` argument is used for two purposes in the `is_valid()`function:

1. We read off the minimum number of realisations for an update - that could just be passed as an integer?
2. [**Most important**]: the `analysis_config` object contains a collection of analysis modules, previously this was a "large" collection of modules - now it is only two and they have distinct roles as being iterable and not iterable. When the object actually implementing the behavior is hidden between several layers of indirection (`analysis_config->default_module->module_config`) the code is less explicit than necessary.

**Comment by sregales-TNO** [comment]
> I think the signature of the `is_valid()` function should get some love and attention before it is exported to Python. It is in particular the `analysis_config` argument which I think should be replaced removed. Currently the `analysis_config` argument is used for two purposes in the `is_valid()`function:
> 
> 1. We read off the minimum number of realisations for an update - that could just be passed as an integer?
> 2. [**Most important**]: the `analysis_config` object contains a collection of analysis modules, previously this was a "large" collection of modules - now it is only two and they have distinct roles as being iterable and not iterable. When the object actually implementing the behavior is hidden between several layers of indirection (`analysis_config->default_module->module_config`) the code is less explicit than necessary.

I agree with your statement. Was trying to do the refactoring of the validation function while making the tests, but was persuaded that it wasn't the place/issue to do so. So I will give the function some love on Monday.

**Comment by oyvindeide** [comment]
Underlying issue solved by: https://github.com/equinor/ert/pull/2962 so will close this.

**Pull request closed without merging**